packages/broker/src/helpers/withTimeout.ts

```diff
@@ -0,0 +1,16 @@
+export const withTimeout = async <T>(promise: Promise<T>, timeout: number): Promise<T> => {
+    return new Promise(async (resolve, reject) => {
+        const timer = setTimeout(() => {
+            reject(new Error('Timeout'))
```

Comment:
I've found that in practice, it helps a bit giving some context in these error messages whilst e.g. debugging.
```suggestion
            reject(new Error('withTimeout: Timeout'))
```

Reply:
Fixed. Didn't change the error message as it is written to the end-user log. But now `withTimeout` throws a `GenericError` which contains an error code (e.g. `NAT_ANALYSIS_TIMEOUT` in this case).